pkg/controllers/replica-controller/handler.go

```diff
@@ -257,7 +261,22 @@ func (c *CStorVolumeReplicaController) cVREventHandler(
 		if isCVRCreateStatus(cvrObj) {
 			return c.cVRAddEventHandler(cvrObj, fullVolName)
 		}
-		return c.getCVRStatus(cvrObj)
+
+		if cvrObj.Annotations[restoreCompletedAnnotation] == "true" {
+			targetIp, err := volumereplica.GetTargetIp(fullVolName)
+			if err != nil {
+				return "", err
+			}
+
+			if targetIp != cvrObj.Spec.TargetIP {
```

Comment:
We can directly set the `targetIP` without verifying whether targetIP already exists or not because `cStor` will return an error is VolumeReplica/zvol already holds a different targetIP(Setting the same targetIP again and again will not return any errors). If we still need to check then changing into below will be good 
```suggestion
			if targetIp == "" {
```